src/mpi/coll/src/csel.c

```diff
@@ -288,7 +299,9 @@ static void validate_tree(csel_node_s * node)
     if (node->type == CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_ANY ||
         node->type == CSEL_NODE_TYPE__OPERATOR__AVG_MSG_SIZE_ANY ||
         node->type == CSEL_NODE_TYPE__OPERATOR__TOTAL_MSG_SIZE_ANY ||
-        node->type == CSEL_NODE_TYPE__OPERATOR__COUNT_ANY) {
+        node->type == CSEL_NODE_TYPE__OPERATOR__COUNT_ANY ||
+        node->type == CSEL_NODE_TYPE__OPERATOR__COMM_AVG_PPN_ANY) {
```

Comment:
The `_ANY` are always for the last "else"-branch, right? I wonder do we need all these tags or can we simply replace them with a `CSEL_NODE_TYPE__OPERATOR__ELSE`.

Just a comment. We can do this in a separate PR.

Reply:
I agree. That would be concise.